BUILD.bazel

```diff
@@ -981,24 +938,263 @@ ray_cc_library(
+    name = "cluster_resource_manager",
+    srcs = ["src/ray/raylet/scheduling/cluster_resource_manager.cc"],
+    hdrs = ["src/ray/raylet/scheduling/cluster_resource_manager.h"],
+    deps = [
+        ":local_resource_manager",
         ":ray_common",
+        "//src/ray/protobuf:gcs_cc_proto",
+        "//src/ray/util:container_util",
+        "//src/ray/util:logging",
+        "@com_google_absl//absl/container:flat_hash_map",
+        "@com_google_absl//absl/container:flat_hash_set",
+        "@com_google_googletest//:gtest_prod",
+    ],
+)
+
+ray_cc_library(
+    name = "cluster_resource_scheduler",
+    srcs = ["src/ray/raylet/scheduling/cluster_resource_scheduler.cc"],
+    hdrs = ["src/ray/raylet/scheduling/cluster_resource_scheduler.h"],
+    deps = [
+        ":cluster_resource_manager",
+        ":composite_scheduling_policy",
+        ":scheduler_internal",
+        "//src/ray/protobuf:gcs_cc_proto",
+        "//src/ray/util:logging",
+        "@com_google_absl//absl/container:flat_hash_map",
+        "@com_google_absl//absl/container:flat_hash_set",
+        "@com_google_googletest//:gtest_prod",
+    ],
+)
+
+ray_cc_library(
+    name = "cluster_task_manager",
+    srcs = [
+        "src/ray/raylet/scheduling/cluster_task_manager.cc",
+        "src/ray/raylet/scheduling/scheduler_stats.cc",
+    ],
+    hdrs = [
+        "src/ray/raylet/scheduling/cluster_task_manager.h",
+        "src/ray/raylet/scheduling/scheduler_stats.h",
+    ],
+    deps = [
+        ":cluster_resource_scheduler",
+        ":cluster_task_manager_interface",
+        ":local_task_manager_interface",
+        ":scheduler_internal",
+        ":scheduler_resource_reporter",
         ":stats_lib",
-        "//src/ray/protobuf:common_cc_proto",
-        "//src/ray/util",
-        "@boost//:asio",
-        "@boost//:system",
-        "@com_github_jupp0r_prometheus_cpp//pull",
-        "@com_google_absl//absl/base:core_headers",
+        "//src/ray/util:logging",
+        "@com_google_absl//absl/container:flat_hash_map",
         "@com_google_absl//absl/container:flat_hash_set",
-        "@com_google_absl//absl/memory",
+    ],
+)
+
+ray_cc_library(
+    name = "cluster_task_manager_interface",
+    hdrs = ["src/ray/raylet/scheduling/cluster_task_manager_interface.h"],
+    deps = [
+        ":rpc_server_call",
+        "//src/ray/protobuf:node_manager_cc_proto",
+    ],
+)
+
+ray_cc_library(
+    name = "local_task_manager_interface",
+    hdrs = ["src/ray/raylet/scheduling/local_task_manager_interface.h"],
+    deps = [
+        ":ray_common",
+        ":scheduler_internal",
+        "@com_google_absl//absl/container:flat_hash_map",
+        "@com_google_absl//absl/container:flat_hash_set",
+    ],
+)
+
+ray_cc_library(
+    name = "local_resource_manager",
+    srcs = ["src/ray/raylet/scheduling/local_resource_manager.cc"],
+    hdrs = ["src/ray/raylet/scheduling/local_resource_manager.h"],
+    deps = [
+        ":gcs_client_lib",
+        ":ray_common",
```

Comment:
Hi could you please paste the TODO (and also the github issue) everywhere, so we don't miss?